src/benchmark_mediation.py

```diff
@@ -617,174 +619,309 @@ def alternative_estimator(y, t, m, x, regularization=True):
-            of linear regression
+    clip : float, default=0.01
+        Limit to clip p_x and f_mtx for numerical stability (min=clip, max=1-clip)
 
-    crossfit integer, default 0
-             number of folds for cross-fitting
+    regularization : boolean, default=True
+        Whether to use regularized models (logistic or linear regression).
+        If True, cross-validation is used to chose among 8 potential
+        log-spaced values between 1e-5 and 1e5
 
-    clip    float
-            limit to clip for numerical stability (min=clip, max=1-clip)
+    calibration : boolean, default=True
+        [...]
+
+    calib_method : str, default="sigmoid"
+        [...]
 
-    regularization boolean, default True
-                   whether to use regularized models (logistic or
-                   linear regression). If True, cross-validation is used
-                   to chose among 8 potential log-spaced values between
-                   1e-5 and 1e5
 
+    Returns
+    -------
+    list
+        A list of estimated effects :
+        [total effect,
+        direct effect on the exposed,
+        direct effect on the unexposed,
+        indirect effect on the exposed,
+        indirect effect on the unexposed,
+        number of clipped samples]
+
+
+    Raises
+    ------
+    ValueError
+        - If t or y are multidimensional.
+        - If x, t, m, or y don't have the same length.
+        - If m is not binary.
     """
-    if regularization:
-        alphas = ALPHAS
-        cs = ALPHAS
-    else:
-        alphas = [0.0]
-        cs = [np.inf]
+    # Format checking
+    if len(y) != len(y.ravel()):
+        raise ValueError("Multidimensional y is not supported")
+    if len(t) != len(t.ravel()):
+        raise ValueError("Multidimensional t is not supported")
+    if len(m) != len(m.ravel()):
+        raise ValueError("Multidimensional m is not supported")
+
     n = len(y)
-    ind = np.arange(n)
     if len(x.shape) == 1:
-        x = x.reshape(-1, 1)
+        x.reshape(n, 1)
     if len(m.shape) == 1:
-        mr = m.reshape(-1, 1)
+        m.reshape(n, 1)
+
+    dim_m = m.shape[1]
+    if n * dim_m != sum(m.ravel() == 1) + sum(m.ravel() == 0):
+        raise ValueError("m is not binary")
+
+    y = y.ravel()
+    t = t.ravel()
+    m = m.ravel()
+    if n != len(x) or n != len(m) or n != len(t):
+        raise ValueError("Inputs don't have the same number of observations")
+
+    # Initialisation
+    (
+        p_x,  # P(T=1|X)
+        f_00x,  # f(M=0|T=0,X)
+        f_01x,  # f(M=0|T=1,X)
+        f_10x,  # f(M=1|T=0,X)
+        f_11x,  # f(M=1|T=1,X)
+        f_m0x,  # f(M|T=0,X)
+        f_m1x,  # f(M|T=1,X)
+        mu_t1,  # E(Y|T=1,M,X)
+        mu_t0,  # E(Y|T=0,M,X)
+        mu_t1m1,  # E(Y|T=1,M=1,X)
+        mu_t1m0,  # E(Y|T=1,M=0,X)
+        mu_t0m1,  # E(Y|T=0,M=1,X)
+        mu_t0m0,  # E(Y|T=0,M=0,X)
+        mumut0t0x,  # E(E(Y|T=0,M,X)|T=0,X)
+        mumut0t1x,  # E(E(Y|T=0,M,X)|T=1,X)
```

Comment:
Using underscores would be much more readable.
Code is meant to be read ...

Reply:
done, [2308342](https://github.com/judithabk6/med_bench/pull/35/commits/23083428d5db757e1afe7098fc95519a46de53a0)